examples/advanced/edge/jobs/cifar10_job.py

```diff
@@ -0,0 +1,89 @@
+
+import argparse
+
+from nvflare.app_opt.pt.file_model_persistor import PTFileModelPersistor
+from nvflare.edge.aggregators.model_update_dxo_factory import ModelUpdateDXOAggrFactory
+from nvflare.edge.assessors.model_update import ModelUpdateAssessor
+from nvflare.edge.edge_job import EdgeJob
+from nvflare.edge.models.model import Cifar10ConvNet
+from nvflare.edge.widgets.evaluator import GlobalEvaluator
+
+
+def export_job(args):
+    job = EdgeJob(
+        name=args.job_name,
+        edge_method="cnn",
+    )
+
+    factory = ModelUpdateDXOAggrFactory()
+    if args.simulation_config_file:
+        job.configure_client(
+            aggregator_factory=factory,
+            max_model_versions=args.max_model_aggr,
+            update_timeout=300.0,
+            simulation_config_file=args.simulation_config_file,
+        )
+    else:
+        job.configure_client(
+            aggregator_factory=factory,
+            max_model_versions=args.max_model_aggr,
+            update_timeout=300.0,
+        )
+
+    evaluator = GlobalEvaluator(
+        model_path="nvflare.edge.models.model.Cifar10ConvNet",
+        torchvision_dataset={"name": "CIFAR10", "path": "/tmp/nvflare/datasets/cifar10"},
+    )
+    job.to_server(evaluator, id="evaluator")
+
+    persistor = PTFileModelPersistor(model=Cifar10ConvNet())
+    persistor_id = job.to_server(persistor, id="persistor")
+
+    assessor = ModelUpdateAssessor(
+        persistor_id=persistor_id,
+        max_model_version=args.max_model_version,
+        max_model_history=args.max_model_history,
+        num_updates_for_model=args.num_updates_for_model,
+        device_selection_size=args.device_selection_size,
+        min_hole_to_fill=args.min_hole_to_fill,
+        device_reuse=args.device_reuse,
+        const_selection=args.const_selection,
+    )
+    job.configure_server(
+        assessor=assessor,
+    )
+
+    job.export_job("/tmp/nvflare/jobs/")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Export CIFAR10 EdgeJob")
+    parser.add_argument("--job_name", help="Name of the job to export")
+    parser.add_argument("--simulation_config_file", help="Path to the simulation config file, leaf job if provided")
+    parser.add_argument("--max_model_aggr", type=int, default=3, help="Maximum number of model aggregations")
+    parser.add_argument(
+        "--max_model_version", type=int, default=10, help="Maximum number of model versions to generate"
+    )
+    parser.add_argument("--max_model_history", type=int, default=1, help="Maximum number of model history to keep")
+    parser.add_argument("--num_updates_for_model", type=int, default=16, help="Number of updates for each model")
+    parser.add_argument(
+        "--device_selection_size", type=int, default=16, help="Number of devices to select for each update"
+    )
+    parser.add_argument("--min_hole_to_fill", type=int, default=16, help="Minimum hole to fill in the model")
```

Comment:
we need to be more descriptive in our documentation to explain this argument, maybe not in this job script.

Reply:
will add when refactoring the device_manager and model_manager